audio.c

```diff
@@ -24,10 +24,20 @@
 #include "audio.h"
 
 int16_t pdmBuffer[2][PDM_BUFFER_LENGTH] = {0};
-int16_t releasedPdmBuffer[PDM_BUFFER_LENGTH] = {0};
+int16_t releasedPdmBuffer[PDM_DEC_BUFFER_LENGTH] = {0};
 static bool fftInputBufferReady = false;
 static int pdmBufferIndex = 0;
 
+void decimate(int16_t* buffer, uint8_t dec_factor)
+{
+  int16_t* dec = (int16_t *) malloc(sizeof(int16_t) * PDM_DEC_BUFFER_LENGTH);
+  for (int i = 0; i < PDM_DEC_BUFFER_LENGTH; i++) {
+      dec[i] = buffer[i*dec_factor];
+  }
+  memcpy(releasedPdmBuffer, dec, sizeof(int16_t) * PDM_DEC_BUFFER_LENGTH);
+  free(dec);
+}
+
```

Comment:
let's change this a bit:

void decimate(int16_t* outputBuffer, int16_t* inputBuffer, uint8_t decimationFactor)

instead of mallocing a new arrow, just pass in the reference to releasedPdmBuffer and directly assign it there.

This will also make it more clear when calling decimate inside pdmEventHandler because right now you have to know that decimate directly modifies releasedPdmBuffer